tf/provider.nix

```
@@ -0,0 +1,26 @@
+{ lib, ... }:
+
+{
+  infra.providers.required = {
+    hcloud = {
+      source = "hetznercloud/hcloud";
+
+      token = lib.tfRef "var.hcloud_token";
+    };
+
+    netlify = {
+      token = lib.tfRef "var.netlify_token";
+      default_team_slug = "prismlauncher";
+    };
+  };
+
+  variable = {
+    hcloud_token = {
+      sensitive = true;
+    };
+
+    netlify_token = {
+      type = lib.literalExpression "string";
```

```suggestion
      type = "string";
```

Otherwise OpenTofu complains with

```
╷
│ Error: Invalid type specification
│ 
│   on config.tf.json line 68, in variable.netlify_token:
│   68:       "type": {
│   69:         "_type": "literalExpression",
│   70:         "text": "string"
│   71:       }
│ 
│ A type specification is either a primitive type keyword (bool, number, string) or a complex type constructor call, like list(string).
╵
```